packages/vite/src/rsc/rscRequestHandler.ts

```diff
@@ -123,9 +126,44 @@ export function createRscRequestHandler() {
       }
 
       try {
+        // surround by performance, but also
+        // TODO (RSC): Add a way to turn this off in production
+        const startedAt = Date.now()
+        const start = performance.now()
         const pipeable = await renderRsc({ rscId, props, rsfId, args })
+        const endedAt = Date.now()
+        const end = performance.now()
+        const duration = end - start
+
         // TODO (RSC): See if we can/need to do more error handling here
         // pipeable.on(handleError)
+
+        // Send to Studio
+        // TODO (RSC): Will only want to do this in dev mode
+
+        // collect render request metadata
+        const metadata = {
+          rsc: {
+            rscId,
+            rsfId,
+            props,
+            args,
+          },
+          request: {
+            basePath,
+            originalUrl: req.originalUrl,
+            url: req.url,
+            headers: req.headers,
+          },
+          performance: {
+            startedAt,
+            endedAt,
+            duration,
+          },
+        }
+
+        createStudioFlightHandler(pipeable as PassThrough, metadata)
```

Comment:
Can you wrap this in a `process.env` check. We do something like that in the mailer code already

Reply:
Looked at mailer:

```ts
    // Mode
    this.mode = this.isTest()
      ? 'test'
      : this.isDevelopment()
        ? 'development'
        : 'production'
```

```ts
  protected isTest() {
    if (this.config.test?.when === undefined) {
      // TODO: Ensure this is a sensible default
      return process.env.NODE_ENV === 'test'
    }
...
```

and

```ts
  protected isDevelopment() {
    if (this.config.development?.when === undefined) {
      // TODO: Ensure this is a sensible default
      return process.env.NODE_ENV !== 'production'
    }
...
```

But can just check envars.

I think I should have some check like:

* isStudio ... has the project setup studio to use
* isDev and isTest ...
* and only send if isStudio and isDev.

And likely construct the performance metadata as well on those conditions.